src/player.cpp

```diff
@@ -3807,6 +3807,11 @@ float player::fine_detail_vision_mod( const tripoint &p ) const
     return std::min( own_light, ambient_light );
 }
 
+int player::get_focus_chance( int focus ) const
+{
+    return roll_remainder( 50 * ( 1 - std::cos( ( M_PI / 100 ) * ( focus % 100 ) ) ) );
+}
```

Comment:
I think you can pull more into this function
```suggestion
int player::get_focus_change( int focus ) const
{
    if( focus >= 100 ) {
        return 1;
    }
    return roll_remainder( 0.5 * ( 1 - std::cos( M_PI / 100 * focus ) ) );
}
```
and update calling code accordingly, e.g. one caller simplifies to just `focus_pool -= get_focus_change( focus_pool )`.